examples/lhotse_usage.py

```diff
@@ -0,0 +1,269 @@
+            subset_split[protocol_split] = sorted(set(sup.recording_id for sup in sups))
+
+    # Create the protocol
+    protocol = LhotseProtocol(
+        recording_set=recording_set,
+        supervision_set=supervision_set,
+        subset_split=subset_split,
+    )
+
+    # Iterate over training files
+    for file in protocol.train():
+        print(f"URI: {file['uri']}")
+        print(f"Annotation: {file['annotation']}")
+        print(f"Annotated regions: {file['annotated']}")
+        # file also contains:
+        # - file['recording']: Lhotse Recording object
+        # - file['supervisions']: Lhotse SupervisionSet
+
+
+def example_with_preprocessors(ami_dir):
+    """Using preprocessors with LhotseProtocol
+
+    Preprocessors allow you to add custom fields to each file
+    that are computed on-the-fly.
+
+    Args:
+        ami_dir: Path to the AMI directory containing Lhotse JSONL files
+    """
+    import lhotse
+    from pathlib import Path
+
+    ami_path = Path(ami_dir)
+
+    # Load train/dev/test recordings and supervisions from the compressed JSONL files
+    # Map from protocol split names to file names
+    split_mapping = {"train": "train", "development": "dev", "test": "test"}
+    file_splits = ["train", "dev", "test"]
+
+    # Load all recordings and supervisions
+    all_recordings = []
+    all_supervisions = []
+
+    for file_split in file_splits:
+        rec_file = ami_path / f"ami-ihm_recordings_{file_split}.jsonl.gz"
+        sup_file = ami_path / f"ami-ihm_supervisions_{file_split}.jsonl.gz"
+
+        if rec_file.exists() and sup_file.exists():
+            print(f"  Loading {file_split}: {rec_file.name} and {sup_file.name}...")
+            all_recordings.extend(lhotse.load_manifest(str(rec_file)))
+            all_supervisions.extend(lhotse.load_manifest(str(sup_file)))
+
+    recording_set = lhotse.RecordingSet.from_recordings(all_recordings)
+    supervision_set = lhotse.SupervisionSet(all_supervisions)
+    print(f"  Loaded {len(recording_set)} recordings and {len(supervision_set)} supervisions")
+
+    # Define train/development/test splits by reading the actual split files
+    subset_split = {}
+    for protocol_split, file_split in split_mapping.items():
+        sup_file = ami_path / f"ami-ihm_supervisions_{file_split}.jsonl.gz"
+        if sup_file.exists():
+            sups = lhotse.load_manifest(str(sup_file))
+            subset_split[protocol_split] = sorted(set(sup.recording_id for sup in sups))
+
+    # Define preprocessors (only use callable preprocessors with Lhotse data)
+    # Note: Template-based preprocessors should reference keys that exist in the file dict
+    preprocessors = {
+        "duration": lambda file: file["recording"].duration,
+        "num_speakers": lambda file: len(set(
+            s.speaker for s in file["supervisions"] if s.speaker is not None
+        )),
+        "audio": lambda file: file["recording"].sources[0].audio,
```

Comment:
right now audio is defined in preprocessor only. 
I am still using lhotse recordings which could allow for a bunch of cool things in preprocessors (e.g. loading even multi channel data).